### PR TITLE
Convert logtime tests

### PR DESCRIPTION
**Pull request by hackebrot**
Convert TestCase `TestLogTime` to `pytest`. #10 

This PR is based on #11 and #12 :exclamation: 

Please note that this PR introduces the plugin **pytest-capturelog**.

### CODE REVIEW

**Comment by The-Compiler** [comment]
Note this PR also contains commits to convert the `qenum_key` and `qflags_key` tests - is that intended?

**Comment by hackebrot** [comment]
Yes I did this on purpose as these PRs convert `test_debug.py`.
I added an according note to the description.

File: qutebrowser/test/utils/debug/test_log_time.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+from qutebrowser.utils import debug
+
+
+def test_log_time(caplog):
+    """Test if log_time logs properly."""
+
+    logger_name = 'qt-tests'
+
+    with caplog.atLevel(logging.DEBUG, logger=logger_name):
+        with debug.log_time(logging.getLogger(), action='foobar'):
```
Comment (hackebrot) [inline qutebrowser/test/utils/debug/test_log_time.py]: I missed `logger_name` :open_mouth:

**Comment by The-Compiler** [comment]
I noticed there's also [pytest-catchlog](https://pypi.python.org/pypi/pytest-catchlog) which is a fork of **pytest-capturelog** (which you probably meant with **caplog** above) - do you know what the difference between those two is? Is there some reason we should use catchlog instead of capturelog?

**EDIT:** Seems like the main difference (apart of some packaging stuff) is [a record_tuples() method](https://github.com/eisensheng/pytest-catchlog/commit/e62b8c2672ddb1d22d46cb2bc55ca387d8ad9133) returning a list of `(logger, level, text)` tuples - does that sound like something useful enough to use the fork instead?

**Comment by hackebrot** [comment]
If there are no significant differences between those two, I'd go with the one that is being used the most. Chances are that potential bugs are spotted sooner. :relaxed:

**Comment by The-Compiler** [comment]
Good point - let's go for `pytest-capturelog` then - 309 github search results, 5 for catchlog.

**Comment by hackebrot** [comment]
If @nicoddemus is fine with these changes, feel free to merge this one! :rocket:

**Comment by nicoddemus** [comment]
Looks good to me. :smile:

**Comment by The-Compiler** [comment]
I did a first run on the CI and this test [failed on Windows](http://qutebrowser.org:8010/builders/win8/builds/441/steps/unittests/logs/stdio).

It seems the assert doesn't actually do the same:

``` python
>>> duration = 0.109316
>>> unittest.TestCase().assertAlmostEqual(duration, 0.1, delta=0.01)
>>> assert round(duration - 0.1, 2) == 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

I think I'll just replace the code by something like:

``` python
assert duration >= 0.09
assert duration <= 0.11
```

Seems much more clear what the test is doing then as well.

**Comment by hackebrot** [comment]
Ok, cool! Go ahead :+1:

**Comment by nicoddemus** [comment]
Note that you can also write this instead:

``` python
assert 0.09 <= duration <= 0.11
```

**Pull request merged**